### PR TITLE
Decluster window

### PR DESCRIPTION
**Pull request by rcgee**
In this PR I add the capability to specify a `time_cutoff` parameter that can optionally be applied to the declustering time window. If `time_cutoff` is specified, the declustering time window is cut, and outside of the window an event of any magnitude is no longer identified as a cluster. Using this feature typically decreases the number of events that are removed from the catalogue during declustering.

`time_cutoff` is specified (in days) in the declustering configuration as so:

```
declust_config = {"time_distance_window": GardnerKnopoffWindow(), 
                  "fs_time_prop": 1.0, 
                  "time_cutoff": 100}
```
This feature was requested by SwissRe

### CODE REVIEW

**Comment by mmpagani** [comment]
Robin, I know you did tests externally but it would be also nice to have a test similar to this one https://github.com/gem/oq-engine/blob/master/openquake/hmtk/tests/seismicity/declusterer/dec_gardner_knopoff_test.py

File: openquake/hmtk/seismicity/declusterer/distance_time_windows.py [hunk 1]
```diff
@@ -55,19 +55,32 @@
...
 
 
+def time_window_cutoff(sw_time, time_cutoff):
+    """
+    Allows for cutting the declustering time window at a specific time, outside
+    of which an event of any magnitude is no longer identified as a cluster
+    """
+    sw_time = np.array(
+        [(time_cutoff / 364.75) if x > (time_cutoff / 364.75)
+            else x for x in sw_time])
```
Comment (micheles) [inline openquake/hmtk/seismicity/declusterer/distance_time_windows.py]: You could introduce a constant with a meaningful name for the number 364.75.

File: openquake/hmtk/tests/seismicity/declusterer/dec_gardner_knopoff_test.py [hunk 2]
```diff
@@ -85,3 +85,20 @@ def test_dec_gardner_knopoff(self):
...
+                  'fs_time_prop' : 1.0,
+                  'time_cutoff' : 100}
+        # Instantiate the declusterer and process the sample catalogue
+        dec = GardnerKnopoffType1()
+        vcl, flagvector = dec.decluster(self.cat, config)
+        print('vcl:', vcl)
+        catalog_flag = self.cat.data['flag']
+        catalog_flag[4] = 0 # event becomes mainshock when time_cutoff = 100
+        print('flagvector:', catalog_flag)
+        self.assertTrue(np.allclose(flagvector,self.cat.data['flag']))
```
Comment (micheles) [inline openquake/hmtk/tests/seismicity/declusterer/dec_gardner_knopoff_test.py]: Never ever user `assertTrue`! There is `assert_allclose` in numpy

File: openquake/hmtk/seismicity/declusterer/distance_time_windows.py [hunk 3]
```diff
@@ -53,6 +53,7 @@
 from openquake.baselib.general import CallableDict
 
 TIME_DISTANCE_WINDOW_FUNCTIONS = CallableDict()
+DAYS = DAYS
```
Comment (micheles) [inline openquake/hmtk/seismicity/declusterer/distance_time_windows.py]: ??

Comment (rcgee) [reply openquake/hmtk/seismicity/declusterer/distance_time_windows.py]: yikes! this was the result of the "find and replace" of 364.75 with DAYS

File: openquake/hmtk/seismicity/declusterer/dec_gardner_knopoff.py [hunk 4]
```diff
@@ -99,9 +99,15 @@ def decluster(self, catalogue, config):
...
-            config['time_distance_window'].calc(catalogue.data['magnitude']))
         # Initial Position Identifier
+        if config.get('time_cutoff'):
+            sw_space, sw_time = (
+                config['time_distance_window'].calc(
+                    catalogue.data['magnitude'], config['time_cutoff']))
+        else:
+            sw_space, sw_time = (
+                config['time_distance_window'].calc(
+                    catalogue.data['magnitude']))
```
Comment (micheles) [inline openquake/hmtk/seismicity/declusterer/dec_gardner_knopoff.py]: Why not
```python
sw_space, sw_time = (
￼                 config['time_distance_window'].calc(
￼                     catalogue.data['magnitude']), config.get('time_cutoff')))
```

**Comment by micheles** [comment]
LGTM, but please update the changelog

**Pull request merged**